### PR TITLE
SWARM-1139 / SWARM-1168 / SWARM-1156 - Logging/stderr/printStackTrace.

### PR DESCRIPTION
**Pull request by bobmcwhirter**
Motivation
----------

Our logging is sub-par, many times just println()'ing.  We also are not
handling exceptions in a decent manner, just printing stack traces.

Modifications
-------------

Split WFSWARM into more logging prefixes and categories.  Renumber
(BREAKING CHANGE if you grep logs), avoid simple printStackTrace().

Modify a few SPI interfaces to allow "throws Exception" so that we
can actually percolate errors up instead of blindly trundling forward
when a customizer has an issue.

Result
------

Cleaning logs, better logging, better error-handling.

- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java [hunk 1]
```diff
@@ -240,16 +230,17 @@ public void deploy(Archive<?> deployment) throws DeploymentException {
                 preparer.prepareArchive(deployment);
             }
 
-            if (this.debug) {
+            if (DeployerMessages.MESSAGES.isDebugEnabled()) {
+                DeployerMessages.MESSAGES.deploying(deployment.getName());
                 Map<ArchivePath, Node> ctx = deployment.getContent();
                 for (Map.Entry<ArchivePath, Node> each : ctx.entrySet()) {
-                    System.err.println(each.getKey() + " // " + each.getValue());
+                    DeployerMessages.MESSAGES.deploymentContent(each.getKey().toString());
```
Comment (kenfinnigan) [inline core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java:237]: Does this need to output key and value or just key now?

File: core/spi/src/main/java/org/wildfly/swarm/spi/api/ArtifactLookup.java [hunk 2]
```diff
@@ -38,12 +38,8 @@ static ArtifactLookup get() {
...
             try {
                 return (ArtifactLookup) Class.forName("org.wildfly.swarm.internal.ArtifactManager").newInstance();
-            } catch (InstantiationException e1) {
-                e1.printStackTrace();
-            } catch (IllegalAccessException e1) {
-                e1.printStackTrace();
-            } catch (ClassNotFoundException e1) {
-                e1.printStackTrace();
+            } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e1) {
+                // TODO handle error
```
Comment (kenfinnigan) [inline core/spi/src/main/java/org/wildfly/swarm/spi/api/ArtifactLookup.java:42]: Guessing this was missed?

**Review by kenfinnigan — APPROVED** [review]
Few comments i've added, but awesome stuff!

File: core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java (hunk 1 above)
Comment (bobmcwhirter) [reply core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java:237]: The value was always crap like "InputStreamAsset" or "ByteArrayAsset"

Comment (bobmcwhirter) [reply core/container/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java:237]: or actually just NodeImpl.

**Pull request merged**